src/pgwire/src/protocol.rs

```diff
@@ -970,15 +975,14 @@ where
             }
 
             let row_desc = portal.desc.relation_desc.clone();
-
             match &mut portal.state {
                 PortalState::NotStarted => {
                     // Start a transaction if we aren't in one. Postgres does this both here and
                     // in bind. We don't do it in bind because I'm not sure what purpose it would
                     // serve us (i.e., I'm not aware of a pgtest that would differ between us and
                     // Postgres).
                     self.start_transaction(Some(1));
-
+                    let execute_started = Instant::now();
```

Comment:
Could `execute` generate and return this instead? Then you could have like `Ok((response, execute_started))` below which feels slightly better? I kinda like making it consistent when the measurement starts instead of having all callers need to determine that for themselves.